rke/templates/kube-router.go

```diff
@@ -0,0 +1,479 @@
+      name: kube-router-context
+    current-context: kube-router-context
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    k8s-app: kube-router
+    tier: node
+  name: kube-router
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      k8s-app: kube-router
+      tier: node
+  template:
+    metadata:
+      labels:
+        k8s-app: kube-router
+        tier: node
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "20243"
+    spec:
+      priorityClassName: system-node-critical
+      serviceAccountName: kube-router
+      serviceAccount: kube-router
+      containers:
+      - name: kube-router
+        image: {{.CNIImage}}
+        imagePullPolicy: Always
+        args:
+        - "--run-router=true"
+        - "--run-firewall={{.RunFirewall}}"
+        - "--run-service-proxy={{.RunServiceProxy}}"
+        - "--kubeconfig=/var/lib/kube-router/kubeconfig"
+        - "--bgp-graceful-restart" 
+        - "--metrics-port=20243"
+        env:
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        - name: KUBE_ROUTER_CNI_CONF_FILE
+          value: /etc/cni/net.d/10-kuberouter.conflist
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 20244
+          initialDelaySeconds: 10
+          periodSeconds: 3
+        resources:
+          requests:
+            cpu: 250m
+            memory: 250Mi
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: lib-modules
+          mountPath: /lib/modules
+          readOnly: true
+        - name: cni-conf-dir
+          mountPath: /etc/cni/net.d
+        - name: kubeconfig
+          mountPath: /var/lib/kube-router
+          readOnly: true
+      initContainers:
+      - name: install-cni
+        image: alpine
+        imagePullPolicy: Always
+        command:
+        - /bin/sh
+        - -c
+        - set -e -x;
+          if [ ! -f /etc/cni/net.d/10-kuberouter.conflist ]; then
+            if [ -f /etc/cni/net.d/*.conf ]; then
+              rm -f /etc/cni/net.d/*.conf;
+            fi;
+            TMP=/etc/cni/net.d/.tmp-kuberouter-cfg;
+            cp /etc/kube-router/cni-conf.json ${TMP};
+            mv ${TMP} /etc/cni/net.d/10-kuberouter.conflist;
+          fi;
+          if [ ! -f /var/lib/kube-router/kubeconfig ]; then
+            TMP=/var/lib/kube-router/.tmp-kubeconfig;
+            cp /etc/kube-router/kubeconfig ${TMP};
+            mv -f ${TMP} /var/lib/kube-router/kubeconfig;
+          fi;
+          mkdir -p /opt/cni/bin;
+          if [ ! "$(ls -A /opt/cni/bin)" ]; then
+            wget https://github.com/containernetworking/plugins/releases/download/v0.8.3/cni-plugins-linux-amd64-v0.8.3.tgz -O /tmp/cni-plugins-linux-amd64-v0.8.3.tgz && tar -xf /tmp/cni-plugins-linux-amd64-v0.8.3.tgz -C /opt/cni/bin/;
```

Comment:
This is provided by rke-tools, why this fallback (which will also break airgap but in general, what is the fallback good for as if binaries are not provided it should just not work)

Reply:
Well.. I didn't know that it was included in rke-tools :thinking: 
Maybe back when I started this PR, it wasn't as I think I copied most things from existing CNI-Plugins. Anyway, if it's included there, I'm going to remove it here :+1: